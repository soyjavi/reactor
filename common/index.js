import { LAYOUT } from './layout/index';
import SHAPE from './shape';
import THEME from './theme';

export { LAYOUT, SHAPE, THEME };
