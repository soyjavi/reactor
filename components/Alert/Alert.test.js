import React from 'react';
import renderer from 'react-test-renderer';

import { THEME } from '../../common';
import { Alert } from '..';

const accept = 'Accept';
const cancel = 'Cancel';
const caption = '... of the printing and typesetting industry';
const fn = () => {};
const icon = 'comment-question-outline';
const onClose = () => {};
const DEFAULTS = {
  title: 'Lorem Ipsum is simply dummy text...',
  visible: true,
};
const {
  COLOR: { BRAND: color },
} = THEME;

jest.useFakeTimers();

describe('<Alert>', () => {
  it('renders', () => {
    const tree = renderer.create(<Alert {...DEFAULTS} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {accept}', () => {
    const tree = renderer.create(<Alert {...DEFAULTS} accept={accept} onAccept={fn} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {cancel}', () => {
    const tree = renderer.create(<Alert {...DEFAULTS} cancel={cancel} onAccept={fn} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {caption}', () => {
    const tree = renderer.create(<Alert {...DEFAULTS} caption={caption} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {color}', () => {
    const tree = renderer.create(<Alert {...DEFAULTS} color={color} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {visible} (false)', () => {
    const tree = renderer.create(<Alert {...DEFAULTS} visible={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('hook:styler', () => {
    const snapshot = renderer.create(<Alert {...DEFAULTS} marginBottom="M" />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
