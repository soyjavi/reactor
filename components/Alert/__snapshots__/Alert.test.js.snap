// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert> hook:styler 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "height": "100%",
      "left": 0,
      "maxHeight": "100%",
      "position": "absolute",
      "top": 0,
      "width": "100%",
      "zIndex": 2,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "backgroundColor": "rgba(0,0,0,0.5)",
          "flex": 1,
          "justifyContent": "center",
        },
        undefined,
        undefined,
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "maxHeight": "100%",
            "position": "absolute",
            "width": "100%",
          },
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        pointerEvents="auto"
        style={Object {}}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
                "padding": 24,
              },
              Array [
                Object {
                  "marginBottom": 16,
                },
                undefined,
              ],
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": undefined,
                  "textAlignVertical": "center",
                },
                Object {
                  "fontSize": 24,
                },
                undefined,
                Object {
                  "fontWeight": "700",
                },
                undefined,
                Object {
                  "marginBottom": 16,
                },
              ]
            }
          >
            Lorem Ipsum is simply dummy text...
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "marginTop": 24,
              }
            }
          >
            <View
              accessible={true}
              focusable={false}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              pointerEvents="none"
              style={
                Array [
                  undefined,
                  Array [
                    Object {
                      "alignContent": "center",
                      "alignItems": "center",
                      "alignSelf": "flex-start",
                      "justifyContent": "center",
                      "width": "auto",
                    },
                    undefined,
                    Object {
                      "backgroundColor": "#ff8c5f",
                      "borderRadius": 0,
                    },
                    undefined,
                    Object {
                      "height": 48,
                      "paddingHorizontal": 16,
                    },
                    Object {
                      "flex": 1,
                    },
                  ],
                ]
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderRadius": 0,
                      "bottom": 0,
                      "left": 0,
                      "overflow": "hidden",
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    0,
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#ffffff",
                      "height": 0,
                      "opacity": 0,
                      "width": 0,
                      "zIndex": 0,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;

exports[`<Alert> renders 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "height": "100%",
      "left": 0,
      "maxHeight": "100%",
      "position": "absolute",
      "top": 0,
      "width": "100%",
      "zIndex": 2,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "backgroundColor": "rgba(0,0,0,0.5)",
          "flex": 1,
          "justifyContent": "center",
        },
        undefined,
        undefined,
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "maxHeight": "100%",
            "position": "absolute",
            "width": "100%",
          },
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        pointerEvents="auto"
        style={Object {}}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
                "padding": 24,
              },
              Array [
                undefined,
              ],
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": undefined,
                  "textAlignVertical": "center",
                },
                Object {
                  "fontSize": 24,
                },
                undefined,
                Object {
                  "fontWeight": "700",
                },
                undefined,
                Object {
                  "marginBottom": 16,
                },
              ]
            }
          >
            Lorem Ipsum is simply dummy text...
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "marginTop": 24,
              }
            }
          >
            <View
              accessible={true}
              focusable={false}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              pointerEvents="none"
              style={
                Array [
                  undefined,
                  Array [
                    Object {
                      "alignContent": "center",
                      "alignItems": "center",
                      "alignSelf": "flex-start",
                      "justifyContent": "center",
                      "width": "auto",
                    },
                    undefined,
                    Object {
                      "backgroundColor": "#ff8c5f",
                      "borderRadius": 0,
                    },
                    undefined,
                    Object {
                      "height": 48,
                      "paddingHorizontal": 16,
                    },
                    Object {
                      "flex": 1,
                    },
                  ],
                ]
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderRadius": 0,
                      "bottom": 0,
                      "left": 0,
                      "overflow": "hidden",
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    0,
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#ffffff",
                      "height": 0,
                      "opacity": 0,
                      "width": 0,
                      "zIndex": 0,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;

exports[`<Alert> when {accept} 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "height": "100%",
      "left": 0,
      "maxHeight": "100%",
      "opacity": 1,
      "position": "absolute",
      "top": 0,
      "width": "100%",
      "zIndex": 2,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "backgroundColor": "rgba(0,0,0,0.5)",
          "flex": 1,
          "justifyContent": "center",
        },
        undefined,
        undefined,
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "maxHeight": "100%",
            "position": "absolute",
            "width": "100%",
          },
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
                "padding": 24,
              },
              Array [
                undefined,
              ],
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": undefined,
                  "textAlignVertical": "center",
                },
                Object {
                  "fontSize": 24,
                },
                undefined,
                Object {
                  "fontWeight": "700",
                },
                undefined,
                Object {
                  "marginBottom": 16,
                },
              ]
            }
          >
            Lorem Ipsum is simply dummy text...
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "marginTop": 24,
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onLayout={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              pointerEvents="box-only"
              style={
                Array [
                  undefined,
                  Array [
                    Object {
                      "alignContent": "center",
                      "alignItems": "center",
                      "alignSelf": "flex-start",
                      "justifyContent": "center",
                      "width": "auto",
                    },
                    undefined,
                    Object {
                      "backgroundColor": "#ff8c5f",
                      "borderRadius": 0,
                    },
                    undefined,
                    Object {
                      "height": 48,
                      "paddingHorizontal": 16,
                    },
                    Object {
                      "flex": 1,
                    },
                  ],
                ]
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": undefined,
                        "textAlignVertical": "center",
                      },
                      Object {
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                      undefined,
                      false,
                      Object {
                        "color": "#ffffff",
                      },
                      Array [
                        Object {
                          "fontWeight": "700",
                          "paddingHorizontal": 8,
                        },
                        false,
                      ],
                    ]
                  }
                >
                  Accept
                </Text>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderRadius": 0,
                      "bottom": 0,
                      "left": 0,
                      "overflow": "hidden",
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    0,
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#ffffff",
                      "height": 0,
                      "opacity": 0,
                      "width": 0,
                      "zIndex": 0,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;

exports[`<Alert> when {cancel} 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "height": "100%",
      "left": 0,
      "maxHeight": "100%",
      "position": "absolute",
      "top": 0,
      "width": "100%",
      "zIndex": 2,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "backgroundColor": "rgba(0,0,0,0.5)",
          "flex": 1,
          "justifyContent": "center",
        },
        undefined,
        undefined,
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "maxHeight": "100%",
            "position": "absolute",
            "width": "100%",
          },
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        pointerEvents="auto"
        style={Object {}}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
                "padding": 24,
              },
              Array [
                undefined,
              ],
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": undefined,
                  "textAlignVertical": "center",
                },
                Object {
                  "fontSize": 24,
                },
                undefined,
                Object {
                  "fontWeight": "700",
                },
                undefined,
                Object {
                  "marginBottom": 16,
                },
              ]
            }
          >
            Lorem Ipsum is simply dummy text...
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "marginTop": 24,
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onLayout={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              pointerEvents="box-only"
              style={
                Array [
                  undefined,
                  Array [
                    Object {
                      "alignContent": "center",
                      "alignItems": "center",
                      "alignSelf": "flex-start",
                      "justifyContent": "center",
                      "width": "auto",
                    },
                    undefined,
                    Object {
                      "backgroundColor": "#ff8c5f",
                      "borderRadius": 0,
                    },
                    undefined,
                    Object {
                      "height": 48,
                      "paddingHorizontal": 16,
                    },
                    Object {
                      "flex": 1,
                    },
                  ],
                ]
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderRadius": 0,
                      "bottom": 0,
                      "left": 0,
                      "overflow": "hidden",
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    0,
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#ffffff",
                      "height": 0,
                      "opacity": 0,
                      "width": 0,
                      "zIndex": 0,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;

exports[`<Alert> when {caption} 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "height": "100%",
      "left": 0,
      "maxHeight": "100%",
      "opacity": 1,
      "position": "absolute",
      "top": 0,
      "width": "100%",
      "zIndex": 2,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "backgroundColor": "rgba(0,0,0,0.5)",
          "flex": 1,
          "justifyContent": "center",
        },
        undefined,
        undefined,
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "maxHeight": "100%",
            "position": "absolute",
            "width": "100%",
          },
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
                "padding": 24,
              },
              Array [
                undefined,
              ],
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": undefined,
                  "textAlignVertical": "center",
                },
                Object {
                  "fontSize": 24,
                },
                undefined,
                Object {
                  "fontWeight": "700",
                },
                undefined,
                Object {
                  "marginBottom": 16,
                },
              ]
            }
          >
            Lorem Ipsum is simply dummy text...
          </Text>
          <Text
            style={
              Array [
                Object {
                  "fontFamily": undefined,
                  "textAlignVertical": "center",
                },
                Object {
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                undefined,
                false,
                undefined,
                undefined,
              ]
            }
          >
            ... of the printing and typesetting industry
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "marginTop": 24,
              }
            }
          >
            <View
              accessible={true}
              focusable={false}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              pointerEvents="none"
              style={
                Array [
                  undefined,
                  Array [
                    Object {
                      "alignContent": "center",
                      "alignItems": "center",
                      "alignSelf": "flex-start",
                      "justifyContent": "center",
                      "width": "auto",
                    },
                    undefined,
                    Object {
                      "backgroundColor": "#ff8c5f",
                      "borderRadius": 0,
                    },
                    undefined,
                    Object {
                      "height": 48,
                      "paddingHorizontal": 16,
                    },
                    Object {
                      "flex": 1,
                    },
                  ],
                ]
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderRadius": 0,
                      "bottom": 0,
                      "left": 0,
                      "overflow": "hidden",
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    0,
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#ffffff",
                      "height": 0,
                      "opacity": 0,
                      "width": 0,
                      "zIndex": 0,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;

exports[`<Alert> when {color} 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "height": "100%",
      "left": 0,
      "maxHeight": "100%",
      "position": "absolute",
      "top": 0,
      "width": "100%",
      "zIndex": 2,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "backgroundColor": "rgba(0,0,0,0.5)",
          "flex": 1,
          "justifyContent": "center",
        },
        undefined,
        undefined,
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "maxHeight": "100%",
            "position": "absolute",
            "width": "100%",
          },
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        pointerEvents="auto"
        style={Object {}}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
                "padding": 24,
              },
              Array [
                undefined,
              ],
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": undefined,
                  "textAlignVertical": "center",
                },
                Object {
                  "fontSize": 24,
                },
                undefined,
                Object {
                  "fontWeight": "700",
                },
                undefined,
                Object {
                  "marginBottom": 16,
                },
              ]
            }
          >
            Lorem Ipsum is simply dummy text...
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "marginTop": 24,
              }
            }
          >
            <View
              accessible={true}
              focusable={false}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              pointerEvents="none"
              style={
                Array [
                  undefined,
                  Array [
                    Object {
                      "alignContent": "center",
                      "alignItems": "center",
                      "alignSelf": "flex-start",
                      "justifyContent": "center",
                      "width": "auto",
                    },
                    undefined,
                    Object {
                      "backgroundColor": "#62e6ac",
                      "borderRadius": 0,
                    },
                    undefined,
                    Object {
                      "height": 48,
                      "paddingHorizontal": 16,
                    },
                    Object {
                      "flex": 1,
                    },
                  ],
                ]
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderRadius": 0,
                      "bottom": 0,
                      "left": 0,
                      "overflow": "hidden",
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    0,
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#ffffff",
                      "height": 0,
                      "opacity": 0,
                      "width": 0,
                      "zIndex": 0,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;

exports[`<Alert> when {visible} (false) 1`] = `
<View
  pointerEvents="none"
  style={
    Object {
      "height": "100%",
      "left": 0,
      "maxHeight": "100%",
      "opacity": 0,
      "position": "absolute",
      "top": 0,
      "width": "100%",
      "zIndex": 2,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "backgroundColor": "rgba(0,0,0,0.5)",
          "flex": 1,
          "justifyContent": "center",
        },
        undefined,
        undefined,
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "maxHeight": "100%",
            "position": "absolute",
            "width": "100%",
          },
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "transform": Array [
              Object {
                "translateY": 907.1200000000001,
              },
            ],
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
                "padding": 24,
              },
              Array [
                undefined,
              ],
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": undefined,
                  "textAlignVertical": "center",
                },
                Object {
                  "fontSize": 24,
                },
                undefined,
                Object {
                  "fontWeight": "700",
                },
                undefined,
                Object {
                  "marginBottom": 16,
                },
              ]
            }
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "marginTop": 24,
              }
            }
          >
            <View
              accessible={true}
              focusable={false}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              pointerEvents="none"
              style={
                Array [
                  undefined,
                  Array [
                    Object {
                      "alignContent": "center",
                      "alignItems": "center",
                      "alignSelf": "flex-start",
                      "justifyContent": "center",
                      "width": "auto",
                    },
                    undefined,
                    Object {
                      "backgroundColor": "#ff8c5f",
                      "borderRadius": 0,
                    },
                    undefined,
                    Object {
                      "height": 48,
                      "paddingHorizontal": 16,
                    },
                    Object {
                      "flex": 1,
                    },
                  ],
                ]
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderRadius": 0,
                      "bottom": 0,
                      "left": 0,
                      "overflow": "hidden",
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    0,
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#ffffff",
                      "height": 0,
                      "opacity": 0,
                      "width": 0,
                      "zIndex": 0,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
